cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -64,6 +66,14 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.MasterC
 		errs = append(errs, err)
 	}
 
+	if err := pkiutil.BackupAPIServerCertAndKey(); err != nil {
```

Comment:
As you said in the TODO: this should happen if version is larger than v1.9.0 _and_ the cert is more than 6 months old